bleak/backends/bluezdbus/manager.py

```diff
@@ -472,12 +477,28 @@ async def passive_scan(
             )
             self._device_removed_callbacks.append(device_removed_callback_and_state)
 
+            # Once a monitoring job is activated by BlueZ, the client can expect to get notified
+            # on the targeted advertisements no matter if there is an ongoing discovery session
+            # (started/stopped with org.bluez.Adapter1.StartDiscovery/StopDiscovery, as done when
+            # using active scanning).
+            # That is why discovery_stopped_callback is not added to self._discovery_stopped_callbacks
+            # at this point, as org.bluez.Adapter1.Discovering status is not relevant for passive
+            # scanning.
+
+            # If advertisement monitor is released before the scanning is stopped, it means that the
+            # kernel does not support passive scanning and error was returned when trying to execute
+            # MGMT command "Add Adv Patterns Monitor" (see https://github.com/hbldh/bleak/issues/1136).
+            # Otherwise, monitor will be activated and start to receive advertisement packets.
+            monitor_processed = asyncio.Queue()
+
             try:
-                monitor = AdvertisementMonitor(filters, discovery_stopped_callback)
+                monitor = AdvertisementMonitor(filters, monitor_processed.put_nowait)
```

Comment:
We can also keep the `discovery_stopped_callback` functionality, if required, by adding on intermediate callback which will call `put_nowait` and then `discovery_stopped_callback`.